### PR TITLE
Better tracking of valid dynamic scissors & viewports

### LINKED ISSUE #798: Core validation mishandles CmdSetViewport / CmdSetScissor which only update a subset of the viewports or scissors.
Currently we assume that CmdSetViewport/CmdSetScissor are replacing /all/ the viewports or scissors, and completely replace our internal representation.

This leads to bogus validation errors at draw time.

### PR DESCRIPTION
**Pull request by chrisforbes**
Fixes #798

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -6583,8 +6605,7 @@ CmdSetViewport(VkCommandBuffer commandBuffer, uint32_t firstViewport, uint32_t v
     if (pCB) {
         skip_call |= addCmd(dev_data, pCB, CMD_SETVIEWPORTSTATE, "vkCmdSetViewport()");
         pCB->status |= CBSTATUS_VIEWPORT_SET;
-        pCB->viewports.resize(viewportCount);
-        memcpy(pCB->viewports.data(), pViewports, viewportCount * sizeof(VkViewport));
+        pCB->viewportMask |= (1u<<viewportCount) - 1u << firstViewport;
```
Comment (tobine) [inline layers/core_validation.cpp]: OSX Android compile error:

`error: operator '<<' has lower precedence than '-'; '-' will be evaluated first`

`note: place parentheses around the '-' expression to silence this warning`

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -6600,8 +6621,7 @@ CmdSetScissor(VkCommandBuffer commandBuffer, uint32_t firstScissor, uint32_t sci
     if (pCB) {
         skip_call |= addCmd(dev_data, pCB, CMD_SETSCISSORSTATE, "vkCmdSetScissor()");
         pCB->status |= CBSTATUS_SCISSOR_SET;
-        pCB->scissors.resize(scissorCount);
-        memcpy(pCB->scissors.data(), pScissors, scissorCount * sizeof(VkRect2D));
+        pCB->scissorMask |= (1u<<scissorCount) - 1u << firstScissor;
```
Comment (tobine) [inline layers/core_validation.cpp]: OSX Android compile error:

`error: operator '<<' has lower precedence than '-'; '-' will be evaluated first`

`note: place parentheses around the '-' expression to silence this warning`

File: layers/core_validation.cpp (hunk 1 above)
Comment (lentinem) [inline layers/core_validation.cpp]: I support clear order of operations with () when possible. In order words I'd prefer () even without the warning.

**Comment by lentinem** [comment]
+1

**Comment by chrisforbes** [comment]
Fixed and pushed to master

**Pull request merged**